invenio_records_resources/records/api.py

```diff
@@ -63,9 +63,14 @@ def get_by_key(cls, record_id, key):
                 return cls(obj.data, model=obj)
 
     @classmethod
-    def list_by_record(cls, record_id):
+    def list_by_record(cls, record_id, with_deleted=False):
         """List all record files by record ID."""
-        for obj in cls.model_cls.query.filter(cls.model_cls.record_id == record_id):
+        query = cls.model_cls.query.filter(cls.model_cls.record_id == record_id)
+
+        if not with_deleted:
+            query = query.filter(cls.model_cls.is_deleted != True)
```

Comment:
is it possible to revert the logic to avoid the double negation? (unless it was done for a reason)

```suggestion
        if with_deleted:
            query = query.filter(cls.model_cls.is_deleted == True)
```